### PR TITLE
[Resolve #337] Fix Integration test when not on Circle CI

### LINKED ISSUE #337: Fix Integration test when not on Circle CI
When not on circle-CI UUIDs for bucket names, were creating invalid bucket names.
To avoid this and generate the bucket name based on account alias or account id. Re-use existing buckets to prevent creating a mess.

### PR DESCRIPTION
**Pull request by JuanCanham**
When not on circle-CI UUIDs for bucket names, were creating invalid bucket names.
To avoid this and generate the bucket name based on account alias or account id.
And re-use existing buckets to prevent creating a mess.

-----------------

* [x] Wrote [good commit messages][1].
* [N/A] [Squashed related commits together][2] after the changes have been approved.
* [x] Commit message starts with `[Resolve #issue-number]` (if a related issue exists).
* [N/A] Added unit tests.
* [N/A] Added integration tests (if applicable).
* [N/A] All unit tests (`make test`) are passing.
* [x] Used the same coding conventions as the rest of the project.
* [x] The new code doesn't generate flake8 (`make lint`) offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

### CODE REVIEW

File: integration-tests/environment.py [hunk 1]
```diff
@@ -54,3 +52,33 @@ def after_all(context):
...
+    if account_aliases:
+        account_alias = account_aliases[0]
+        return '{}-{}'.format(
+            bucket_prefix[:62 - len(account_alias)],
+            account_alias
+        )
+
+    sts = boto3.client("sts")
+    account_number = sts.get_caller_identity()['Account']
+    return '{}-{}'.format(bucket_prefix, account_number)
```
Comment (theseanything) [inline integration-tests/environment.py]: Could we not reduce the whole function to lines 58, 82,83,84 ?

File: integration-tests/environment.py [hunk 2]
```diff
@@ -54,3 +52,33 @@ def after_all(context):
...
+        bucket['Name'] for bucket in s3.list_buckets()['Buckets']
+        if bucket['Name'].startswith(bucket_prefix)
+    ]
+
+    if existing_buckets:
+        return existing_buckets[0]
+
+    iam = boto3.client("iam")
+    account_aliases = iam.list_account_aliases()['AccountAliases']
+    if account_aliases:
```
Comment (theseanything) [inline integration-tests/environment.py]: Why this over just using number?

File: integration-tests/environment.py [hunk 3]
```diff
@@ -54,3 +52,33 @@ def after_all(context):
     context.project_code = "sceptre-integration-tests"
     context.bucket_name = "sceptre-integration-tests-templates"
     update_config(context)
+
+
+def get_integration_test_bucket_name():
+    bucket_prefix = "sceptre-integration-tests-templates"
+
+    if os.environ.get("CIRCLECI"):
```
Comment (theseanything) [inline integration-tests/environment.py]: Don't really need this we can just switch to the new format

**Comment by theseanything** [comment]
Thanks @JuanCanham  - I agree with idea, not sure if it need to be that complex though?

File: integration-tests/environment.py [hunk 4]
```diff
@@ -54,3 +52,33 @@ def after_all(context):
...
+        return bucket_prefix
+
+    s3 = boto3.client('s3')
+
+    existing_buckets = [
```
Comment (theseanything) [inline integration-tests/environment.py]: Don't need this if you are always using account number

**Comment by JuanCanham** [comment]
Updated and simplified

**Comment by ngfgrant** [comment]
@JuanCanham were these changes intended for V1 or V2?

**Comment by JuanCanham** [comment]
V2, I can make a backport branch with stuff that my project would benefit from in v1

although tbh integration tests working in v1 would be nice too

**Pull request merged**